### PR TITLE
Feature/jr 558 display occupation titles according to gender

### PR DESCRIPTION
**Pull request by michael-wirth**

### CODE REVIEW

File: src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts [hunk 1]
```diff
@@ -84,13 +84,15 @@ export class CandidateDetailComponent implements OnInit {
...
-            .combineLatest(currentLanguage$)
-            .map(([experiences, lang]) => experiences.map(this.formatOccupationLabel()))
-            .map((experiences) => experiences
-                .filter((experience) => experience.wanted)
-                .sort((a, b) => +b.lastJob - +a.lastJob));
+            .flatMap((profile: CandidateProfile) =>
+                Observable.of(profile.jobExperiences)
+                    .flatMap((experiences) => Observable.combineLatest(experiences.map(this.enrichWithLabels.bind(this))))
+                    .combineLatest(currentLanguage$)
+                    .map(([experiences, lang]) => experiences.map(this.formatOccupationLabel(profile.gender)))
```
Comment (michael-wirth) [inline src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts]: Is there a better way to get the gender as in a nested `flapMap` method?

File: src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts [hunk 2]
```diff
@@ -108,11 +110,10 @@ export class CandidateDetailComponent implements OnInit {
                 Object.assign({}, jobExperience, { occupationLabels }));
     }
 
-    private formatOccupationLabel(): (jobExperience: EnrichedJobExperience) => EnrichedJobExperience {
+    private formatOccupationLabel(gender: Gender): (jobExperience: EnrichedJobExperience) => EnrichedJobExperience {
         return (jobExperience: EnrichedJobExperience) => {
             const { male, female } = jobExperience.occupationLabels;
-            const occupation = male + ((female && male !== female) ? ' / ' + female : '');
-
+            const occupation = (gender.toString() === 'FEMALE' && female) ? female : male;
```
Comment (michael-wirth) [inline src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts]: It seems that `gender` is defined as `enum`, but the real value is a `string`, do we need some custom mapping?

File: src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts (hunk 1 above)
Comment (birom) [reply src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts]: We can slightly change the original version:
```
   this.jobExperiences$ = this.candidateProfile$
            .map((profile: CandidateProfile) => profile.jobExperiences)
            .flatMap((experiences) => Observable.combineLatest(experiences.map(this.enrichWithLabels.bind(this))))
            .combineLatest(this.candidateProfile$, currentLanguage$)
            .map(([experiences, profile, lang]) => experiences.map(this.formatOccupationLabel(profile.gender)))
            .map((experiences) => experiences
                .filter((experience) => experience.wanted)
                .sort((a, b) => +b.lastJob - +a.lastJob));
```

File: src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts (hunk 2 above)
Comment (birom) [reply src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts]: I think this the mapping should be implemented in the CandidateService. 
Jhipster generates a few helper methodes for dates by default (eg.: convertResponse and convertItemFromServer in the JobService). We could add these and extend them with the enum mapping.

**Comment by michael-wirth** [comment]
fixed

File: src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts (hunk 2 above)
Comment (michael-wirth) [reply src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts]: fixed

File: src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts (hunk 1 above)
Comment (michael-wirth) [reply src/main/webapp/app/candidate-search/candidate-detail/candidate-detail.component.ts]: fixed

**Pull request merged**